xcpmd/src/acpi-events.c

```diff
@@ -30,133 +30,406 @@
+        }
+
+        if (sscanf(tokens[2], "%x", &type) != 1) {
+            xcpmd_log(LOG_DEBUG, "ACPI type field doesn't look like a hex integer: %s\n", tokens[2]);
+            return;
+        }
+        if (sscanf(tokens[3], "%x", &data) != 1) {
+            xcpmd_log(LOG_DEBUG, "ACPI data field doesn't look like a hex integer: %s\n", tokens[3]);
+            return;
+        }
+
+        if (type == ACPI_AC_NOTIFY_STATUS)
+            handle_ac_adapter_event(data);
+
     }
+    else if (!strcmp(class, ACPI_BUTTON_CLASS)) {
 
-    if ( strstr(acpi_buffer, "video") )
-    {
-        /* Special HP case, check the device the notification is for */
-        if ( (pm_quirks & PM_QUIRK_SW_ASSIST_BCL_HP_SB) &&
-             (strstr(acpi_buffer, "DD02") == NULL) )
+        if (subclass == NULL) {
+            xcpmd_log(LOG_DEBUG, "Button event with null subclass\n");
             return;
+        }
+
+
+        if (!strcmp(subclass, ACPI_BUTTON_SUBCLASS_LID)) {
+
+            if (tokens[2] == NULL)
+                data = LID_UNKNOWN;
+            else if (!strcmp(tokens[2], "open"))
+                data = LID_OPEN;
+            else if (!strcmp(tokens[2], "close"))
+                data = LID_CLOSED;
+            else
+                data = LID_UNKNOWN;
+
+            handle_lid_event(data);
+        }
+        else if (!strcmp(subclass, ACPI_BUTTON_SUBCLASS_POWER)) {
+            handle_power_button_event();
+        }
+        else if (!strcmp(subclass, ACPI_BUTTON_SUBCLASS_SLEEP)) {
+            handle_sleep_button_event();
+        }
+        else if (!strcmp(subclass, ACPI_BUTTON_SUBCLASS_SUSPEND)) {
+            handle_suspend_button_event();
+        }
+        else
+            xcpmd_log(LOG_DEBUG, "Received unknown button subclass: %s\n", subclass);
+    }
+    else if (!strcmp(class, ACPI_VIDEO_CLASS)) {
 
-        if ( strstr(acpi_buffer, "00000086") )
-        {
+        if (subclass == NULL) {
+            handle_video_event();
+        }
+        else if (!strcmp(subclass, ACPI_VIDEO_SUBCLASS_BRTUP)) {
             handle_bcl_event(BCL_UP);
         }
-        else if ( strstr(acpi_buffer, "00000087") )
-        {
+        else if (!strcmp(subclass, ACPI_VIDEO_SUBCLASS_BRTDN)) {
             handle_bcl_event(BCL_DOWN);
         }
+        else if (!strcmp(subclass, ACPI_VIDEO_SUBCLASS_BRTCYCLE)) {
+            handle_bcl_event(BCL_CYCLE);
+        }
+        else if (!strcmp(subclass, ACPI_VIDEO_SUBCLASS_TABLETMODE)) {
+
+            if (tokens[3] == NULL) {
+                xcpmd_log(LOG_DEBUG, "Tablet mode event with null data field\n");
+                return;
+            }
+            if (sscanf(tokens[3], "%x", &data) != 1) {
+                xcpmd_log(LOG_DEBUG, "Tablet mode data field doesn't look like a hex integer: %s\n", tokens[3]);
+                return;
+            }
+
+            handle_tablet_mode_event(data);
+        }
     }
+    //else
+    //    xcpmd_log(LOG_DEBUG, "Unhandled ACPI class: %s\n", class);
```

Comment:
This was clogging up /var/log/messages something fierce, and not providing any useful debug information at this time. I can go ahead and remove this one.